File: sdks/typescript/test-local.ts
import { AIGateClient, AIGateError } from './src';

async function main() {
  // Replace with your actual project ID from the dashboard
  const PROJECT_ID = process.env.AI_GATE_PROJECT_ID || 'your-project-id-here';

  console.log('🚀 Testing AI Gate TypeScript SDK\n');
  console.log('Project ID:', PROJECT_ID);

  const client = new AIGateClient({
    projectId: PROJECT_ID,
    debug: true,
  });

  try {
    console.log('\n📝 Creating session...\n');

    const session = await client.createSession({
      voice: 'echo',
      temperature: 0.7,
      instructions: 'You are a helpful assistant for testing the SDK',
    });

    console.log('\n✅ SUCCESS! Session created:\n');
    console.log('  Session ID:', session.session_id);
    console.log('  OpenAI Session:', session.id);
    console.log('  Model:', session.metadata.model);
    console.log('  Voice:', session.metadata.voice);
    console.log('  TURN Server:', session.metadata.turn_server);
    console.log('  Created:', session.metadata.created_at);

    if (session.client_secret) {
      console.log('\n🔐 Client Secret (for WebRTC):');
      console.log('  ', session.client_secret.value.substring(0, 30) + '...');
    }

    console.log('\n🌐 TURN Credentials:');
    session.turn_credentials.iceServers.forEach((server, i) => {
      console.log(`  Server ${i + 1}:`, server.urls);
    });

    console.log('\n✅ SDK is working correctly!');
  } catch (error) {
    if (error instanceof AIGateError) {
      console.error('\n❌ AI Gate Error:');
      console.error('  Message:', error.message);
      console.error('  Status Code:', error.statusCode);

      if (error.statusCode === 402) {
        console.error('\n💡 Quota exceeded. Upgrade your plan:');
        console.error('  ', (error.response as any)?.upgrade_url);
      } else if (error.statusCode === 429) {
        console.error('\n💡 Rate limit or concurrent limit exceeded.');
        console.error('  Please wait and try again.');
      } else if (error.statusCode === 401) {
        console.error('\n💡 Authentication failed.');
        console.error('  Check that your project ID is correct.');
      }
    } else {
      console.error('\n❌ Unexpected error:', error);
    }
    process.exit(1);
  }
}

main();

File: sdks/python/test_local.py
#!/usr/bin/env python3
"""Test script for AI Gate Python SDK"""

import os
from ai_gate import AIGateClient, AIGateError


def main():
    # Replace with your actual project ID from the dashboard
    project_id = os.getenv("AI_GATE_PROJECT_ID", "your-project-id-here")

    print("🚀 Testing AI Gate Python SDK\n")
    print(f"Project ID: {project_id}")

    with AIGateClient(project_id=project_id, debug=True) as client:
        try:
            print("\n📝 Creating session...\n")

            session = client.create_session(
                voice="echo",
                temperature=0.7,
                instructions="You are a helpful assistant for testing the SDK",
            )

            print("\n✅ SUCCESS! Session created:\n")
            print(f"  Session ID: {session['session_id']}")
            print(f"  OpenAI Session: {session['id']}")
            print(f"  Model: {session['metadata']['model']}")
            print(f"  Voice: {session['metadata']['voice']}")
            print(f"  TURN Server: {session['metadata']['turn_server']}")
            print(f"  Created: {session['metadata']['created_at']}")

            if "client_secret" in session:
                print("\n🔐 Client Secret (for WebRTC):")
                print(f"  {session['client_secret']['value'][:30]}...")

            print("\n🌐 TURN Credentials:")
            for i, server in enumerate(session["turn_credentials"]["iceServers"]):
                print(f"  Server {i + 1}: {server['urls']}")

            print("\n✅ SDK is working correctly!")

        except AIGateError as error:
            print("\n❌ AI Gate Error:")
            print(f"  Message: {error}")
            print(f"  Status Code: {error.status_code}")

            if error.status_code == 402:
                print("\n💡 Quota exceeded. Upgrade your plan:")
                if error.response:
                    print(f"  {error.response.get('upgrade_url')}")
            elif error.status_code == 429:
                print("\n💡 Rate limit or concurrent limit exceeded.")
                print("  Please wait and try again.")
            elif error.status_code == 401:
                print("\n💡 Authentication failed.")
                print("  Check that your project ID is correct.")

            exit(1)

        except Exception as error:
            print(f"\n❌ Unexpected error: {error}")
            exit(1)


if __name__ == "__main__":
    main()
